mkdocs/cli.py

```diff
@@ -120,13 +123,15 @@ def json_command(clean, config_file, strict, site_dir):
 @cli.command(name="gh-deploy")
 @click.option('--clean', is_flag=True, help=clean_help)
 @click.option('--config-file', type=click.File('rb'), help=config_file_help)
-def gh_deploy_command(config_file, clean):
+@click.option('--message', '-m', help=commit_message_help)
+@click.option('--branch', '-b', help=remote_branch_help)
+def gh_deploy_command(config_file, clean, message, branch):
     """Deply your documentation to GitHub Pages"""
     config = load_config(
         config_file=config_file
     )
     build.build(config, clean_site_dir=clean)
-    gh_deploy.gh_deploy(config)
+    gh_deploy.gh_deploy(config, message=message, branch=branch)
```

Comment:
Rather than passing the parameter here, it would be passed to load_config (see the other commands, this overwrites the config with CLI parameters and keeps them consistent).